sycl/include/CL/sycl/detail/pi.h

```diff
@@ -389,25 +464,45 @@ const pi_queue_properties PI_QUEUE_ON_DEVICE_DEFAULT =
 typedef _pi_result pi_result;
 typedef _pi_platform_info pi_platform_info;
 typedef _pi_device_type pi_device_type;
+typedef _pi_device_mem_cache_type pi_device_mem_cache_type;
+typedef _pi_device_local_mem_type pi_device_local_mem_type;
 typedef _pi_device_info pi_device_info;
+typedef _pi_program_info pi_program_info;
 typedef _pi_context_info pi_context_info;
 typedef _pi_queue_info pi_queue_info;
 typedef _pi_image_info pi_image_info;
+typedef _pi_kernel_info pi_kernel_info;
+typedef _pi_kernel_group_info pi_kernel_group_info;
+typedef _pi_event_info pi_event_info;
+typedef _pi_command_type pi_command_type;
 typedef _pi_mem_type pi_mem_type;
+typedef _pi_mem_advice pi_mem_advice;
 typedef _pi_image_channel_order pi_image_channel_order;
 typedef _pi_image_channel_type pi_image_channel_type;
 typedef _pi_buffer_create_type pi_buffer_create_type;
 typedef _pi_sampler_addressing_mode pi_sampler_addressing_mode;
 typedef _pi_sampler_filter_mode pi_sampler_filter_mode;
 typedef _pi_sampler_info pi_sampler_info;
 typedef _pi_event_status pi_event_status;
-typedef _pi_event_info pi_event_info;
-typedef _pi_command_type pi_command_type;
-typedef _pi_program_info pi_program_info;
 typedef _pi_program_build_info pi_program_build_info;
 typedef _pi_program_build_status pi_program_build_status;
 typedef _pi_kernel_info pi_kernel_info;
-typedef _pi_kernel_group_info pi_kernel_group_info;
+
+// For compatibility with OpenCL define this not as enum.
+using pi_device_partition_property = intptr_t;
+static constexpr pi_device_partition_property PI_DEVICE_PARTITION_EQUALLY =
```

Comment:
@smaslov-intel, can we use C++ in this header?
If so, I suggest keep the coding style consistent and replace all `typedef` with `using`.
Why we have two separate files: pi.h and pi.hpp?

Reply:
done.